lib/main.dart

```diff
@@ -1,129 +1,214 @@
-                            ]));
-                      }))
-        ]));
+                  ),
+                )),
+            StreamBuilder<SearchMetadata>(
+              stream: _searchMetadata,
+              builder: (context, snapshot) {
+                if (!snapshot.hasData) {
+                  return const SizedBox.shrink();
+                }
+                return Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: Text('${snapshot.data!.nbHits} hits'),
+                );
+              },
+            ),
+            Expanded(
+              child: _hits(context),
+            )
+          ],
+        ),
+      ),
+    );
   }
 
-  @override
-  void initState() {
-    super.initState();
-    _textFieldController.addListener(() {
-      if (_searchText != _textFieldController.text) {
-        setState(() {
-          _searchText = _textFieldController.text;
-        });
-        _getSearchResult(_searchText);
-      }
-    });
-    _getSearchResult('');
-  }
+  Widget _hits(BuildContext context) => PagedListView<int, Product>(
+      pagingController: _pagingController,
+      builderDelegate: PagedChildBuilderDelegate<Product>(
+          noItemsFoundIndicatorBuilder: (_) => const Center(
+            child: Text('No results found'),
+          ),
+          itemBuilder: (_, item, __) => Container(
+            color: Colors.white,
+            height: 80,
+            padding: const EdgeInsets.all(8),
+            child: Row(
+              children: [
+                SizedBox(width: 50, child: Image.network(item.image)),
+                const SizedBox(width: 20),
+                Expanded(child: Text(item.name))
+              ],
+            ),
+          )));
+
+  Widget _filters(BuildContext context) => Scaffold(
+    appBar: AppBar(
+      title: const Text('Filters'),
+    ),
+    body: StreamBuilder<List<SelectableItem<Facet>>>(
+        stream: _facetList.facets,
+        builder: (context, snapshot) {
+          if (!snapshot.hasData) {
+            return const SizedBox.shrink();
+          }
+          final selectableFacets = snapshot.data!;
+          return ListView.builder(
+              padding: const EdgeInsets.all(8),
+              itemCount: selectableFacets.length,
+              itemBuilder: (_, index) {
+                final selectableFacet = selectableFacets[index];
+                return CheckboxListTile(
+                  value: selectableFacet.isSelected,
+                  title: Text(
+                      "${selectableFacet.item.value} (${selectableFacet.item.count})"),
+                  onChanged: (_) {
+                    _facetList.toggle(selectableFacet.item.value);
+                  },
+                );
+              });
+        }),
+  );
 
   @override
   void dispose() {
-    _textFieldController.dispose();
+    _searchTextController.dispose();
+    _pagingController.dispose();
```

Comment:
Ideally, we need also to dispose of search components (searcher, filter state, facet list) either one by one or using a `CompositeDisposable` :)

Reply:
Done, I added them one-by-one to keep it simple.